core/pta/attestation.c

```diff
@@ -0,0 +1,600 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Elias von Däniken
+ */
+
+#include <compiler.h>
+#include <stdio.h>
+#include <kernel/pseudo_ta.h>
+#include <mm/tee_pager.h>
+#include <mm/tee_mm.h>
+#include <pta_attestation.h>
+#include <string.h>
+#include <string_ext.h>
+#include <malloc.h>
+#include <tee_api_types.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+
+#define RSA_SIG_SIZE_BYTE 256
+
+/*
+ * Attestation data is static to keep until the device boots again.
+ * Afterwards the attestation data has to be reloaded and everthing
+ * has to be attestated newly.
+ */
+static struct attestation_data {
+	uint8_t *iv;
+	uint32_t iv_size;
+	uint8_t *pub;
+	uint32_t pub_size;
+	uint8_t *enc_priv;
+	uint32_t enc_priv_size;
+	uint8_t *sig;
+	uint32_t sig_size;
+} att_data;
+
+static TEE_Result priv_asn1_decode(uint8_t *priv,
+				uint8_t **d, size_t *d_size)
+{
+
+	uint32_t size;
+	uint32_t offset = 7; // Beginn at the start of the modulus
```

Comment:
We normally don't use C++-style comment in OP-TEE (except for the SPDX tag in `.c` files.
What do others think? Should we accept them now? (they are allowed in the Linux kernel apparently).

Reply:
> We normally don't use C++-style comment in OP-TEE (except for the SPDX tag in `.c` files.
> What do others think? Should we accept them now? (they are allowed in the Linux kernel apparently).

Do you have a link?

Reply:
https://lkml.org/lkml/2019/6/27/117

And there this older (2016) rant from Linus https://lkml.org/lkml/2016/7/8/625 where he says he is favorable to C++-style comments.

Reply:
Speaking for myself here:
I guess I could "tolerate" them in .c files, but I still think they should be avoided.
In my mind those are lazy comments that you add as reminders for yourself to be cleaned up before submitting the patch.
So sure you can use them, but each time I see one I'll be thinking "so you don't have time to write a proper comment?". ;-)

Reply:
@jenswi-linaro +1, the power of habits I suppose :)

Reply:
Yes, I admit it's bit due to habit. But there's at least one technical reason to avoid them. They cannot be used in .h files included by assembly files, so there are places where they are inappropriate. Why not use comments that are acceptable everywhere?

Reply:
Fix [here](https://github.com/OP-TEE/optee_os/pull/4011/commits/eb02698772a7f35c163441f03c7f203f8dc8d2bc)